sql/catalyst/src/main/antlr4/org/apache/spark/sql/catalyst/parser/SqlBase.g4

```diff
@@ -472,11 +472,11 @@ identifierComment
     ;
 
 relationPrimary
-    : tableIdentifier sample? (AS? strictIdentifier)?               #tableName
-    | '(' queryNoWith ')' sample? (AS? strictIdentifier)?           #aliasedQuery
-    | '(' relation ')' sample? (AS? strictIdentifier)?              #aliasedRelation
-    | inlineTable                                                   #inlineTableDefault2
-    | identifier '(' (expression (',' expression)*)? ')'            #tableValuedFunction
+    : tableIdentifier sample? (AS? strictIdentifier)?                                        #tableName
+    | '(' queryNoWith ')' sample? (AS? strictIdentifier)?                                    #aliasedQuery
+    | '(' relation ')' sample? (AS? strictIdentifier)?                                       #aliasedRelation
+    | inlineTable                                                                            #inlineTableDefault2
+    | identifier '(' (expression (',' expression)*)? ')' (AS? identifier identifierList?)?   #tableValuedFunction
```

Comment:
oh, it seems I misunderstood what you pointed out. You meant should we need to support a query like `SELECT * FROM [[tvf]] AS t(a, b, ...)` in this pr? Yea, I know we currently support `range` only as a table value function though, I also think it'd be better to put a more general rule in this file. So +1 for keeping this.